### PR TITLE
Add incomplete interval util function

### PR DESCRIPTION
**Pull request by jalack**

### CODE REVIEW

**Comment by jbwyme** [comment]
can we get some test coverage on this guy?

**Comment by jalack** [comment]
Let me see if there are tests in `analytics` somewhere

**Comment by jbwyme** [comment]
I can bet you that there aren't :/

**Comment by jalack** [comment]
you win the bet

File: lib/util/chart.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/**
+  * isIncompleteInterval -- Returns true if the last time-period has not been completed 
+  *
+  * @param {Hash} series A single series, in highcharts format
+  * @param {Hash} an options hash.
+  *      * (required) 'unit' interval in milliseconds
+  *        OR A string describing the interval ('hour', 'day', 'week', 'month')
+  *      * (required) 'utcOffset' timezone offset in minutes
+  *        OR A string describing the interval ('hour', 'day', 'week', 'month')
+  *      * "adjust_for_local_time" This adjusts the estimation for local time.
```
Comment (jbwyme) [inline lib/util/chart.js]: also let's refactor to camelCase

File: test/util/chart.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    expect(isIncompleteInterval(this.data, { unit: `quarter` })).to.equal(true);
+  });
+
+  it(`returns false when we're looking at a quarter four months ago`, function() {
+    const msInAMonth = 1000 * 60 * 60 * 24 * 30;
+    this.clock.tick(this.lastDate + msInAMonth * 4);
+    expect(isIncompleteInterval(this.data, { unit: `quarter` })).to.equal(false);
+  });
+
+});
```
Comment (jbwyme) [inline test/util/chart.js]: baller! but how come no week or month tests?

File: test/util/highcharts-data.json [hunk 3]
```diff
@@ -0,0 +1,126 @@
+[
```
Comment (jbwyme) [inline test/util/highcharts-data.json]: unless you plan to use this somewhere else you may as well name this file something more explicit like `test/util/chart-fixtures.js`

File: test/util/chart.js (hunk 2 above)
Comment (jalack) [inline test/util/chart.js]: added a couple for each. to be candid, there's a ton of edge cases here involving correctly identifying each period of time at various utc offsets adjusting for local time or not, etc. but I think this suite exercises the major branches and should help catch regressions in the event of a substantial refactor
